java/vortex-jni/src/main/java/dev/vortex/api/ArrayStream.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * (c) Copyright 2025 SpiralDB Inc. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.vortex.api;
+
+import java.io.Closeable;
```

Comment:
AutoCloseable?

Reply:
I wanted to allow these to be held as instance fields so chose Closeable instead (AutoCloseable gets inherited so can still be try-with'ed)

Reply:
you can still hold the value that is autocloseable without having to auto close it